File: techhalo/src/app/common/CartDetail.ts
import { Cart } from "./Cart";
import { Product } from "./Product";
import { PromotionCode } from "./promotions";

export class CartDetail {
    'cartDetailId': number;
    'quantity': number;
    'price': number;
    'discountProductCart': number;
    'product': Product;
    'cart': Cart;
    'promotionCode': PromotionCode | null; // Cho phép giá trị null

    constructor(id: number, quantity: number, price: number, product: Product, cart: Cart) {
        this.cartDetailId = id;
        this.quantity = quantity;
        this.price = price;
        this.product = product;
        this.cart = cart;
    }
}

File: techhalo_Backend/src/main/java/vn/fs/entity/CartDetail.java
package vn.fs.entity;

import java.io.Serializable;

import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@SuppressWarnings("serial")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "cartDetails")
public class CartDetail implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long cartDetailId;
    private int quantity;
    private Double price;
    private Double discountProductCart;

    @OneToOne
    @JoinColumn(name = "promotionId")
    private PromotionCode promotionCode;

    @ManyToOne
    @JoinColumn(name = "productId")
    private Product product;

    @ManyToOne
    @JoinColumn(name = "cartId")
    private Cart cart;

}
